CONTRIBUTING.md

```diff
@@ -0,0 +1,156 @@
+# How to contribute to EquinoxJS
+
+We would love for you to contribute to EquinoxJS and help make it even better
+than it is today! As a contributor, here are the guidelines we would like you
+to follow:
+
+## <a href="question"></a> Got a question or problem?
+
+**Do not open issues for general support questions. Github issues are reserved
+for bug reports and feature requests**. To get your questions answered please
+head over to the [DDD-CQRS-ES discord][discord].
+
+## <a href="bug"></a> Found a bug?
+
+If you find a bug in the source code, you can help us by
+[submitting an issue](#submit-issue) to our [GitHub Repository][github]. Even better, you can
+[submit a Pull Request](#submit-pr) with a fix.
+
+## <a name="feature"></a> Missing a Feature?
+
+You can _request_ a new feature by [submitting an issue](#submit-issue) to our GitHub
+Repository. If you would like to _implement_ a new feature, please submit an issue with
+a proposal for your work first, to be sure that we can use it.
+Please consider what kind of change it is:
+
+- For a **Major Feature**, first open an issue and outline your proposal so
+  that it can be discussed. This will also allow us to better coordinate our
+  efforts, prevent duplication of work, and help you to craft the change so
+  that it is successfully accepted into the project. For your issue name,
+  please prefix your proposal with `[discussion]`, for example "[discussion]:
+  your feature idea".
+- **Small Features** can be crafted and directly [submitted as a Pull
```

Comment:
Should this be reworded to say that it's still always better to open an issue describing what you're up to first? You want to hit a balance between not putting roadblocks and bureaucracy in the way and not frustrating someone who has invested time in writing and testing a change only to be told "no, this is a bad idea; we can't take it" and/or "we feel you should do that via <tiny bit uglier workaround> as this feature is not worth the complexity it adds to the codebase". Ultimately as a maintainer, you'd prefer a small PR with tests every time as it means the request has been validated and there's no need to second guess what's being proposed, but it still hurts on both sides to have to reject a PR, and it would be nice to avoid that.

Reply:
agreed, will reword